common/flatpak-dir.c

```diff
@@ -309,7 +309,19 @@ flatpak_remote_state_ensure_metadata (FlatpakRemoteState *self,
 
       /* If the collection ID is NULL the metadata comes from the summary */
       if (self->metadata_fetch_error != NULL)
-        error_msg = g_strdup (self->metadata_fetch_error->message);
+        {
+          /* Fetch error can be IO_ERROR_CANCELLED in instances like quick searching of apps in software clients.
+           * Hence, avoid converting them to FLATPAK_ERROR from IO_ERROR domain so the clients can handle the
+           * cancellation appropriately. */
+          if (g_error_matches (self->metadata_fetch_error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
```

Comment:
Doesn't the `summary_fetch_error` branch below need the same treatment? How about `flatpak_remote_state_ensure_summary` above? It seems a bit ad-hoc to add this special handling in two places.

Reply:
Yeah, it is adhoc to some extend.
I wasn't able to trigger "summary not found" error-banner :(
But it's LGTM to handle the same case there too.

Actually, one more thing if we can actually teach `flatpak_fail_error` to preserve the `IO_CANCELLED` but I am in dilemma because it can be seen as a non-trivial behavior that the `flatpak_fail_error` utility does

Reply:
Hmm, this is a bit tricky. One of the reasons we do flatpak_fail_error() is that errors are sometimes reported over the system-helper dbus border, and we need to ensure the errors are properly handled like that.

However, a) we don't actually cancel anything in the system-helper atm, and b) I belive the built in glib errors do properly get conveted when sent over dbus, so i think it makes a lot of sense for flatpak_fail_error() to handle IO_CANCELLED.